### PR TITLE
Streamline linting

### LINKED ISSUE #216: Streamline linting Github Workflows
**Is your feature request related to a problem? Please describe.**

Currently all of our linting workflows of the templates are in their own files. There's
run_flake8_linting.yml, run_bandit.yml and run_mlf_core_lint.yml 
Too many files and we're always setting up Python several times.

**Describe the solution you'd like**

What about a single lint.yml file that runs each of these (in parallel)?

### PR DESCRIPTION
**Pull request by Emiller88**
Closes #216 

I was debating caching the things that are `pip installed` also but let me know what y'all think

### CODE REVIEW

**Comment by Zethson** [comment]
@Emiller88 You will need to escape the cookiecutter statements with jinja2 raw statements.
That's why the workflows are failing.

And +1 for caching pip dependencies.

File: mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.8']
+    steps:
+      - uses: actions/checkout@v2
+        name: Check out source-code repository
+
+      - name: Setup Python
+        uses: actions/setup-python@v1
```
Comment (Zethson) [inline mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml:21]:
```suggestion
        uses: actions/setup-python@v2
```

File: mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    types: [published]
+
+jobs:
+
+  mlf-core-lint:
+    name: Run mlf-core lint
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.8']
```
Comment (Zethson) [inline mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml:15]: Think we don't need a matrix here but can run this with just 3.9

**Comment by Emiller88** [comment]
Ah I thought I was being clever. I'll revert it.

File: mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        run: |
+          mlf-core lint .
+
+
+  flake8:
+    name: Run flake8
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.8']
```
Comment (Zethson) [inline mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml:38]: 3.9 only as well

File: mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        run: |
+            flake8 .
+
+  bandit:
+    name: Run bandit
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.head_commit.message, '[ci skip]')"
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.8']
```
Comment (Zethson) [inline mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml:62]: 3.9 only

File: mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    steps:
+      - uses: actions/checkout@v2
+        name: Check out source-code repository
+
+      - name: Setup Python
+        uses: actions/setup-python@v1
```
Comment (Zethson) [inline mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml:68]:
```suggestion
        uses: actions/setup-python@v2
```

File: mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.8']
+    steps:
+      - uses: actions/checkout@v2
+        name: Check out source-code repository
+
+      - name: Setup Python
+        uses: actions/setup-python@v1
```
Comment (Zethson) [inline mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml:44]:
```suggestion
        uses: actions/setup-python@v2
```

**Review by Zethson — CHANGES_REQUESTED** [review]
Small nits

File: mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml [hunk 7]
```diff
@@ -10,17 +10,14 @@ jobs:
...
     steps:
       - uses: actions/checkout@v2
         name: Check out source-code repository
 
       - name: Setup Python
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v2
         with:
-          python-version: {% raw %}${{ matrix.python }}{% endraw %}
+          python-version: '3.9'
```
Comment (Zethson) [inline mlf_core/create/templates/common_all_files/{{cookiecutter.common_all}}/.github/workflows/lint.yml:20]: Think that it doesn't even have to be a string and even 3.9 works already :)

**Comment by Zethson** [comment]
I'll test it real quick. Thanks @Emiller88!

**Review by Zethson — APPROVED** [review]
LGTM. Thanks again @Emiller88 !

**Pull request merged**